### PR TITLE
Switch to using a makefile for builds/tests

### PR DESCRIPTION
**Pull request by anz-gordonj7**
Use a Makefile!

make build test lint coverage release

output from 'make help'
```
$ make help
all                            test, lint then build
build                          Build sysl into the ./dist folder
clean                          Clean temp and build files
coverage                       Verify the test coverage remains high
grammar                        Regenerate the antlr4 grammars
install                        Install the sysl binary into $(GOPATH)/bin
lint                           Run golangci-lint
release                        Build release binaries for all supported platforms into ./release
test                           Run tests without coverage
```

`make` without anything runs all which is the usual developer workflow 

@anz-bank/sysl-developers

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		./cmd/sysl
+
+build: ## Build sysl into the ./dist folder
+	go build -o ./dist/sysl -ldflags="$(LDFLAGS)" -v ./cmd/sysl
+
+deps:
+	go get -v -t -d ./...
+
+.PHONY: release
+release: windows linux darwin ## Build release binaries for all supported platforms into ./release
```
[the remark this answers is not in the document]
Comment (anz-gordonj7) [reply Makefile]: done

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+clean: ## Clean temp and build files
+	rm -rf release dist pkg/grammar/temp
+
+# Autogen rules
+ANTLR = java -jar pkg/lib/antlr-4.7-complete.jar
+GRAMMARS = pkg/grammar/SyslParser.g4 \
+		   pkg/grammar/SyslLexer.g4
+
+antlr: pkg/grammar/SyslParser.g4 pkg/grammar/SyslLexer.g4
```
[the remark this answers is not in the document]
Comment (anz-gordonj7) [reply Makefile]: done

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+else
+	VERSION := $(COMMIT)-$(BRANCH)-$(BUILD_DATE)
+endif
+
+LDFLAGS = $(shell echo -X main.Version=$(VERSION) -X main.GitCommit=$(COMMIT) -X main.BuildDate=$(BUILD_DATE))
+
+all: test lint build coverage ## test, lint, build, coverage test run
+
+
+.PHONY: build lint test coverage
```
Comment (camh-anz) [inline Makefile]: all deps install clean antlr and grammar should be .PHONY too

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+coverage: ## Verify the test coverage remains high
+	./scripts/check-coverage.sh 80
+
+test: ## Run tests without coverage
+	rm -rf pkg/grammar/temp
+	$(TESTEXE)
+
+BINARY := sysl
+PLATFORMS := windows linux darwin
+os = $(word 1, $@)
```
Comment (camh-anz) [inline Makefile]: not sure why you're using `$(word ...)` here. `$@` is the name of the target being built, so it is only one word anyway. I think you should just be able to use `$@` in place of `$(os)` in the recipe below.

Comment (anz-gordonj7) [reply Makefile]: Not gonna lie copied it straight off some blog post :) I've successful avoided makefiles for just about my whole career :)

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+BRANCH ?= $(shell git rev-parse --abbrev-ref HEAD)
+
+# Commit hash from git
+COMMIT=$(shell git rev-parse --short HEAD)
+
+# Tag on this commit
+TAG = $(shell git tag --points-at HEAD)
+
+
+ifneq ("$(which gotestsum)", "")
```
[the remark this answers is not in the document]
Comment (anz-gordonj7) [reply Makefile]: k

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+
+BUILD_DATE=$(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
+ifneq ("$(TAG)", "")
+	VERSION := $(TAG)
+else
+	VERSION := $(COMMIT)-$(BRANCH)-$(BUILD_DATE)
+endif
+
+LDFLAGS = $(shell echo -X main.Version=$(VERSION) -X main.GitCommit=$(COMMIT) -X main.BuildDate=$(BUILD_DATE))
```
[the remark this answers is not in the document]
Comment (anz-gordonj7) [reply Makefile]: fixed

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+ifneq ("$(which gotestsum)", "")
+	TESTEXE := gotestsum --
+else
+	TESTEXE := go test ./...
+endif
+
+
+BUILD_DATE=$(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
+ifneq ("$(TAG)", "")
```
[the remark this answers is not in the document]
Comment (anz-gordonj7) [reply Makefile]: fixed

File: Makefile (hunk 4 above)
Comment (anz-gordonj7) [reply Makefile]: fixed

**Pull request merged**